src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -436,40 +423,100 @@ private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExp
         }
 
         /// <summary>
-        ///  Returns a byte blob that matches serialized content of single array initializer.    
-        ///  returns -1 if the initializer is null or not an array of literals
+        /// Generates a byte blob that matches serialized content of an array.
         /// </summary>
-        private int TryGetRawDataForArrayInit(BoundArrayInitialization initializer, out ImmutableArray<byte> data)
+        /// <returns>True if the blob is generated successfully, false otherwise</returns>
+        private bool TryGetRawDataForArray(BoundArrayCreation initializer, out ImmutableArray<byte> data, out int elementCount)
         {
-            data = default;
+            elementCount = GetArrayElementCount(initializer);
 
-            if (initializer == null)
+            if (elementCount < 0)
+            {
+                data = default;
+                return false;
+            }
+
+            if (elementCount == 0)
+            {
+                data = ImmutableArray<byte>.Empty;
+                return true;
+            }
+
+            var arrayType = (ArrayTypeSymbol)initializer.Type;
+            var elementType = arrayType.ElementType.EnumUnderlyingTypeOrSelf();
+
+            if (initializer.InitializerOpt != null)
+            {
+                return TryGetRawDataForArrayInit(initializer.InitializerOpt, elementType, out data);
+            }
+
+            var elementSize = elementType.SpecialType.SizeInBytes();
```

Comment:
It may be worth leaving a comment: since we're dealing with zeroes, we don't care about endianess and therefore we don't care about element size.

Reply:
I've added a comment about endianness but I haven't got the "element size" part.